server_manager/www/app.ts

```diff
@@ -1035,31 +1038,99 @@ export class App {
     }
   }
 
-  private async refreshServerMetrics(
+  private async refreshServerMetricsUI(
     selectedServer: server_model.Server,
     serverView: ServerView
   ) {
     try {
       const serverMetrics = await selectedServer.getServerMetrics();
 
-      let totalUserHours = 0;
-      let totalAverageDevices = 0;
-      for (const {averageDevices, userHours} of serverMetrics.server) {
-        totalAverageDevices += averageDevices;
-        totalUserHours += userHours;
+      let bandwidthUsageTotal = 0;
+      const bandwidthUsageComparator: Comparator<server_model.ServerMetrics> = (
+        server1,
+        server2
+      ) => server2.dataTransferred.bytes - server1.dataTransferred.bytes;
+      const bandwidthUsageHeap = new Heap(bandwidthUsageComparator);
+
+      let tunnelTimeTotal = 0;
+      const tunnelTimeComparator: Comparator<server_model.ServerMetrics> = (
+        server1,
+        server2
+      ) => server2.tunnelTime.seconds - server1.tunnelTime.seconds;
+      const tunnelTimeHeap = new Heap(tunnelTimeComparator);
+
+      let devicesTotal = 0;
+      const devicesComparator: Comparator<server_model.ServerMetrics> = (
+        server1,
+        server2
+      ) => server2.devices - server1.devices;
+      const devicesHeap = new Heap(devicesComparator);
+
+      for (const server of serverMetrics.server) {
+        bandwidthUsageTotal += server.dataTransferred.bytes;
+        bandwidthUsageHeap.push(server);
+
+        tunnelTimeTotal += server.tunnelTime.seconds;
+        tunnelTimeHeap.push(server);
+
+        devicesTotal += server.devices;
+        devicesHeap.push(server);
       }
 
-      serverView.totalUserHours = totalUserHours;
-      serverView.totalAverageDevices = totalAverageDevices;
+      // support legacy metrics view
+      serverView.totalInboundBytes = bandwidthUsageTotal;
 
-      let totalInboundBytes = 0;
-      for (const {dataTransferred} of serverMetrics.accessKeys) {
-        if (!dataTransferred) continue;
-
-        totalInboundBytes += dataTransferred.bytes;
-      }
-
-      serverView.totalInboundBytes = totalInboundBytes;
+      const NUMBER_OF_ASES_TO_SHOW = 4;
+      serverView.bandwidthUsageTotal = this.formatGigabyteValueAndUnit(
+        bandwidthUsageTotal / BYTES_IN_GIGABYTES
```

Comment:
You should pick the unit based on the value. Otherwise you will end up with zeros